src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -241,3 +448,43 @@ def upload_stream(
             file_name=file_name,
             overwrite=overwrite,
         )
+
+    def get(
+        self,
+        stage_location: str,
+        target_directory: str,
+        options: Dict[str, str] = None,
+    ):
+        if not stage_location.startswith("@"):
+            raise SnowparkSQLException("SQL compilation error")
```

Comment:
We discussed the error behavior last week, let's try to change all SQL specific errors to something more debuggable.

Reply:
updated to use more explicit error message